bundle/src/main/java/com/adobe/acs/commons/wcm/tags/DefineObjects.java

```diff
@@ -0,0 +1,101 @@
+package com.adobe.acs.commons.wcm.tags;
+
+import aQute.bnd.annotation.ProviderType;
+import com.day.cq.commons.LanguageUtil;
+import com.day.cq.wcm.api.components.Component;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.jcr.resource.JcrPropertyMap;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import tldgen.BodyContentType;
+import tldgen.Tag;
+
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+import javax.servlet.jsp.tagext.BodyTagSupport;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This tag is similar to the OOTB cq:defineObjects tag which adds
+ * component instance-level properties to the pageContext of a JSP,
+ * but it instead sets sitewideProperties and mergedProperties maps.
+ *
+ * sitewideProperties contains the site-wide properties for the
+ * current component.
+ *
+ * mergedProperties is a merge of the instance-level and site-wide
+ * properties for the current component, giving preference to an
+ * instance-level property value when a site-wide property exists
+ * with the same name.
+ */
+@ProviderType
+@Tag(bodyContentType = BodyContentType.JSP, value = "defineObjects")
+public class DefineObjects extends BodyTagSupport {
```

Comment:
Please refactor this as a BindingsValuesProvider (which could, in turn, be used by a JSP tag) so that these merged properties can be used in Sightly and other scripting engines.